experimental/yarpl/include/reactivestreams/ReactiveStreams.h

```diff
@@ -0,0 +1,39 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#pragma once
+
+#include <exception>
+#include <memory>
+
+namespace reactivestreams_yarpl {
+// TODO remove 'yarpl' if we agree on this usage elsewhere
+
+class Subscription {
+ public:
+  virtual ~Subscription() = default;
+  virtual void request(int64_t n) = 0;
+  virtual void cancel() = 0;
+};
+
+template <typename T>
+class Subscriber {
+ public:
+  virtual ~Subscriber() = default;
+  virtual void onSubscribe(Subscription*) = 0;
```

Comment:
Passing Subscription as raw pointer feels to me that we will have some memory issues sooner or later. It is hard to manage lifetime with raw pointers. I would suggest using unique_ptr<Subscription>. It will have some consequences to the application code, like it will be impossible to implement a class which implements for Subscriber and Subscription interfaces, but those might be just "healthy" consequences..